titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/KubeApiServerIntegrator.java

```diff
@@ -518,29 +538,44 @@ private void podUpdated(V1Pod pod) {
         lastPodUpdate.put(podName, pod);
 
         if (phase.equalsIgnoreCase(PENDING)) {
-            publishContainerEvent(podName, Launched, REASON_NORMAL, reason, Optional.empty());
+            publishContainerEvent(podName, Launched, REASON_NORMAL, reasonMessage, Optional.empty());
         } else if (phase.equalsIgnoreCase(RUNNING) && pod.getMetadata().getDeletionTimestamp() == null) {
             TitusExecutorDetails titusExecutorDetails = new TitusExecutorDetails(
                     Collections.emptyMap(),
                     new TitusExecutorDetails.NetworkConfiguration(
-                            Boolean.valueOf(pod.getMetadata().getAnnotations().getOrDefault("IsRoutableIp", "true")),
+                            Boolean.parseBoolean(pod.getMetadata().getAnnotations().getOrDefault("IsRoutableIp", "true")),
                             status.getPodIP(),
                             pod.getMetadata().getAnnotations().getOrDefault("EniIpAddress", "UnknownEniIpAddress"),
                             pod.getMetadata().getAnnotations().getOrDefault("EniId", "UnknownEniId"),
                             pod.getMetadata().getAnnotations().getOrDefault("ResourceId", "UnknownResourceId")
                     )
             );
-            publishContainerEvent(podName, StartInitiated, REASON_NORMAL, reason, Optional.of(titusExecutorDetails));
-            publishContainerEvent(podName, Started, REASON_NORMAL, reason, Optional.empty());
+            publishContainerEvent(podName, StartInitiated, REASON_NORMAL, reasonMessage, Optional.of(titusExecutorDetails));
+            publishContainerEvent(podName, Started, REASON_NORMAL, reasonMessage, Optional.empty());
         } else if (phase.equalsIgnoreCase(SUCCEEDED)) {
-            publishContainerEvent(podName, Finished, REASON_NORMAL, reason, Optional.empty());
+            publishContainerEvent(podName, Finished, REASON_NORMAL, reasonMessage, Optional.empty());
         } else if (phase.equalsIgnoreCase(FAILED)) {
-            publishContainerEvent(podName, Finished, REASON_FAILED, reason, Optional.empty());
+            publishContainerEvent(podName, Finished, REASON_FAILED, reasonMessage, Optional.empty());
         }
     }
 
     private void publishContainerEvent(String taskId, TaskState taskState, String reasonCode, String reasonMessage,
```

Comment:
is there anything useful that can be in the old `reasonCode` we may be potentially discarding here?

Reply:
It is mapped to the pod phase so it is not useful at this time unless we add more codes.